packages/manager/apps/pci-object-storage/src/pages/objects/container/object/versions/ShowVersions.page.tsx

```diff
@@ -0,0 +1,258 @@
+
+export type TTags = {
+  [key: string]: string;
+};
+
+export default function ObjectPage() {
+  const { storageId, objectName } = useParams();
+
+  const [searchParams] = useSearchParams();
+  const { data: project } = useProject();
+
+  const { hasMaintenance, maintenanceURL } = useProductMaintenance(
+    project?.project_id,
+  );
+
+  const { formatBytes } = useBytes();
+
+  const hrefProject = useProjectUrl('public-cloud');
+  const { t: tObjects } = useTranslation('objects');
+  const { t: tContainer } = useTranslation('container');
+
+  const objectStorageHref = useHref('..');
+  const containerDetailsHref = useHref(
+    `../${storageId}?region=${searchParams.get('region')}`,
+  );
+
+  const { data: region } = useGetRegion(
+    project?.project_id,
+    searchParams.get('region'),
+  );
+
+  const { storage: targetContainer, storages } = useStorage(
+    project?.project_id,
+    storageId,
+    searchParams.get('region'),
+  );
+
+  const { url } = useStorageEndpoint(project?.project_id, targetContainer);
+
+  const { data: serverContainer } = useServerContainer(
+    project?.project_id,
+    searchParams.get('region'),
+    targetContainer?.name,
+    targetContainer?.id,
+  );
+
+  const container = useMemo((): TContainer => {
+    if (!serverContainer) return undefined;
+    const s3StorageType = targetContainer?.s3StorageType;
+
+    return {
+      ...serverContainer,
+      id: serverContainer?.id || targetContainer?.id,
+      name: serverContainer?.name || targetContainer?.name,
+      objectsCount:
+        serverContainer?.storedObjects || serverContainer?.objectsCount,
+      usedSpace: formatBytes(
+        serverContainer?.storedBytes || serverContainer?.objectsSize,
+        2,
+        1024,
+      ),
+      publicUrl: url,
+      s3StorageType,
+      regionDetails: s3StorageType ? region : undefined,
+      staticUrl: serverContainer?.staticUrl || serverContainer?.virtualHost,
+    };
+  }, [serverContainer, region, targetContainer, url]);
+
+  const {
+    data: ObjectsVersions,
+    isLoading: isObjectsLoading,
+    refetch: refetchObjectVersions,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+  } = useServerContainerObjectVersions({
+    projectId: project?.project_id,
+    region: searchParams.get('region'),
+    name: storageId,
+    key: encodeURIComponent(objectName),
+    isS3StorageType: container?.s3StorageType,
+  });
+
+  useEffect(() => {
+    if (container?.s3StorageType) {
+      refetchObjectVersions();
+    }
+  }, [
+    storageId,
+    searchParams.get('region'),
+    storages,
+    container,
+    searchParams.get('refetch'),
+    refetchObjectVersions,
+  ]);
+
+  const handleFetchNextPage = () => {
+    if (hasNextPage && !isFetchingNextPage) {
+      fetchNextPage();
+    }
+  };
+
+  const containerObjectsWithIndex = useMemo(() => {
+    if (!ObjectsVersions && !container?.s3StorageType) return [];
+    return ObjectsVersions?.map((object, index) => ({
```

Comment:
ObjectVersions cannot be null at this point

Reply:
It may and i had a bug because of it that's I added it

Reply:
@selm3n I guess it's in the line 153: you had already checked if ObjectsVersions is undefined doing `if (!ObjectsVersions && !container?.s3StorageType) return [];`.
So you can modify:
```suggestion
    return ObjectsVersions.map((object, index) => ({
```

Reply:
<img width="688" alt="Screenshot 2025-03-28 at 14 09 50" src="https://github.com/user-attachments/assets/76fc7ac7-d90d-46bc-876c-7c1691891301" />

Reply:
when i remove it i directly get this error